Mage.Sets/src/mage/cards/v/Vault101BirthdayParty.java

```diff
@@ -0,0 +1,119 @@
+package mage.cards.v;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SagaAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.card.AuraCardCanAttachToPermanentId;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.FoodToken;
+import mage.game.permanent.token.HumanWarriorToken;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.TargetPermanent;
+import mage.target.common.TargetCardInLibrary;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author The_REAL_Boudidi
+ */
+public final class Vault101BirthdayParty extends CardImpl {
+
+    public Vault101BirthdayParty(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{W}");
+        
+        this.subtype.add(SubType.SAGA);
+
+        // (As this Saga enters and after your draw step, add a lore counter. Sacrifice after III.)
+        SagaAbility sagaAbility = new SagaAbility(this);
+
+        // I -- Create a 1/1 white Human Soldier creature token and a Food token.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_I,
+                new CreateTokenEffect(new HumanWarriorToken()),
+                new CreateTokenEffect(new FoodToken()));
+
+        // II, III -- You may put an Aura or Equipment card from your hand or graveyard onto the battlefield.
+        // If an Equipment is put onto the battlefield this way, you may attach it to a creature you control.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_II, SagaChapter.CHAPTER_III,
+                new Vault101BirthdayPartyEffect());
+
+        this.addAbility(sagaAbility);
+
+    }
+
+    private Vault101BirthdayParty(final Vault101BirthdayParty card) {
+        super(card);
+    }
+
+    @Override
+    public Vault101BirthdayParty copy() {
+        return new Vault101BirthdayParty(this);
+    }
+
+}
+
+class Vault101BirthdayPartyEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterCard("");
+
+    static {
+        filter.add(Predicates.or(
+                SubType.AURA.getPredicate(),
+                SubType.EQUIPMENT.getPredicate()
+        ));
+    }
+
+    public Vault101BirthdayPartyEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "put an Aura or Equipment card from your hand or graveyard onto the battlefield. "
+                         + "If an Equipment is put onto the battlefield this way, you may attach it to a creature you control." ;
+    }
+
+    private Vault101BirthdayPartyEffect(final mage.cards.v.Vault101BirthdayPartyEffect effect) {

```

Comment:
(don't need the `mage.cards.v`)

Reply:
Thanks for telling me, this is my first attempt at adding a card to XMage.

Reply:
two other locations to clean up